packages/medusa/src/api/routes/admin/stock-locations/delete-stock-location.ts

```diff
@@ -0,0 +1,73 @@
+import { EntityManager } from "typeorm"
+import { IStockLocationService } from "../../../../interfaces"
+
+/**
+ * @oas [delete] /stock-locations/{id}
+ * operationId: "DeleteStockLocationsStockLocation"
+ * summary: "Delete a Stock Location"
+ * description: "Delete a Stock Location"
+ * x-authenticated: true
+ * parameters:
+ *   - (path) id=* {string} The ID of the Stock Location to delete.
+ * x-codeSamples:
+ *   - lang: JavaScript
+ *     label: JS Client
+ *     source: |
+ *       import Medusa from "@medusajs/medusa-js"
+ *       const medusa = new Medusa({ baseUrl: MEDUSA_BACKEND_URL, maxRetries: 3 })
+ *       // must be previously logged in or use api token
+ *       medusa.admin.stockLocations.delete(stock_location_id)
+ *         .then(({ id, object, deleted }) => {
+ *           console.log(id)
+ *         })
+ *   - lang: Shell
+ *     label: cURL
+ *     source: |
+ *       curl --location --request DELETE 'https://medusa-url.com/admin/stock-locations/{id}' \
+ *       --header 'Authorization: Bearer {api_token}'
+ * security:
+ *   - api_token: []
+ *   - cookie_auth: []
+ * tags:
+ *   - StockLocation
+ * responses:
+ *   200:
+ *     description: OK
+ *     content:
+ *       application/json:
+ *         schema:
+ *           type: object
+ *           properties:
+ *             id:
+ *               type: string
+ *               description: The ID of the deleted Stock Location.
+ *             object:
+ *               type: string
+ *               description: The type of the object that was deleted.
+ *               format: stock_location
+ *             deleted:
+ *               type: boolean
+ *               description: Whether or not the Stock Location was deleted.
+ *               default: true
+ *   "400":
+ *     $ref: "#/components/responses/400_error"
+ */
+export default async (req, res) => {
+  const { id } = req.params
+
+  const stockLocationService: IStockLocationService = req.scope.resolve(
+    "stockLocationService"
+  )
+
+  const manager: EntityManager = req.scope.resolve("manager")
+
+  await manager.transaction(async (transactionManager) => {
+    await stockLocationService.delete(id)
+  })
```

Comment:
todo: this should not be in a transaction like this since the `manager` and `stockLocationService` don't share the same DB connection.